### PR TITLE
Add azure front door with default WAF policy

### PR DESCRIPTION
**Pull request by echappen**
Includes a convenience script to add a Front Door resource with related default WAF policy an existing web application.

## Questions
- Currently this script isn't automatically run in iac. Would we want to run this in the full iac script for any web apps, or tack on manually after app deployment?

### CODE REVIEW

**Comment by ryanhofdotgov** [comment]
Assuming it can all be done with `Contributor`-level access, I think we want to set it up immediately following each web app's App Service configuration.

**Comment by ryanhofdotgov** [comment]
@echappen Just to clarify, was this a PR you were wanting to merge in now, or just to get some feedback ahead of integrating into the dashboard and/or query tool?

**Comment by echappen** [comment]
@ryanhofdotgov I'll WIP this while I set this up for immediate application in the main script.

**Comment by echappen** [comment]
@ryanhofdotgov Ready for review

File: iac/create-metrics-resources.bash [hunk 1]
```diff
@@ -256,16 +256,26 @@ EOF
...
+        appName=$DASHBOARD_APP_NAME \
+        servicePlan=$APP_SERVICE_PLAN \
+        metricsApiUri=$metrics_api_uri)
+
+  echo "Create Front Door and WAF policy for dashboard app"
+  ./add-front-door-to-app.bash \
+    $azure_env \
+    $RESOURCE_GROUP \
+    dashboard \
+    $DASHBOARD_HOST_PREFIX.azurewebsites.net
```
Comment (ryanhofdotgov) [inline iac/create-metrics-resources.bash:278]: Need a way to flip between public/commercial cloud and government. On Government, [this is called azurewebsites.us](https://docs.microsoft.com/en-us/azure/azure-government/compare-azure-government-global-azure). 

I had to do something similar for the PostgreSQL connection string:
https://github.com/18F/piipan/blob/6f2c9bad68bbe974ce4b9f21b64233f2a74744d1/iac/iac-common.bash#L60

File: iac/create-resources.bash [hunk 2]
```diff
@@ -636,6 +636,13 @@ main () {
         servicePlan=$APP_SERVICE_PLAN \
         OrchApiUri=$orch_api_uri)
 
+  echo "Create Front Door and WAF policy for query tool app"
+  ./add-front-door-to-app.bash \
+    $azure_env \
+    $RESOURCE_GROUP \
+    querytool \
+    $query_tool_name.azurewebsites.net
```
Comment (ryanhofdotgov) [inline iac/create-resources.bash:644]: Same as above.

**Review by ryanhofdotgov — COMMENTED** [review]
Nice - the security team should be pleased to see this! There's a few minor aspects that need some tweaks, but should be relatively light lifts.

File: iac/create-metrics-resources.bash [hunk 3]
```diff
@@ -256,16 +256,26 @@ EOF
...
-    --name $DASHBOARD_APP_NAME \
-    --resource-group $RESOURCE_GROUP \
-    --template-file ./arm-templates/dashboard-app.json \
-    --parameters \
-      location=$LOCATION \
-      resourceTags="$RESOURCE_TAGS" \
-      appName=$DASHBOARD_APP_NAME \
-      servicePlan=$APP_SERVICE_PLAN \
-      metricsApiUri=$metrics_api_uri
+  DASHBOARD_HOST_PREFIX=$(\
```
[the remark this answers is not in the document]
Comment (echappen) [reply iac/create-metrics-resources.bash:259]: `DASHBOARD_HOST_PREFIX` appends the hash of the resource id, which is made in the template.

File: iac/create-metrics-resources.bash (hunk 1 above)
Comment (echappen) [reply iac/create-metrics-resources.bash:278]: I think it's possible to output the entire host name from the deployment template, which would make a conditional like that moot. But I couldn't figure out how to do that 🤔

File: iac/add-front-door-to-app.bash [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  echo "WAF name: ${waf_full_name}"
+
+  az deployment group create \
+    --name $front_door_full_name \
+    --resource-group $resource_group \
+    --template-file ./arm-templates/front-door-app-service.json \
+    --parameters \
+      appAddress=$app_address \
+      frontDoorHostName="${front_door_full_name}.azurefd.net" \
```
[the remark this answers is not in the document]
Comment (echappen) [reply iac/add-front-door-to-app.bash]: I couldn't find the gov version of the frontdoor domain listed out anywhere either. I [made some functions](https://github.com/18F/piipan/pull/507/commits/8a95c4a775f2f152e14762e5e26fd2cd561ac979#diff-3f860f483e93933e27df096b2bbeea9cd9a0252ad92c104cb073c9cf5740b5dfR74) in iac common to make this switch more explicit.

File: iac/arm-templates/front-door-app-service.json [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            "apiVersion": "2020-07-01",
+            "type": "Microsoft.Network/frontdoors",
+            "name": "[parameters('frontDoorName')]",
+            "tags": "[parameters('resourceTags')]",
+            "location": "[variables('location')]",
+            "dependsOn": [
+                "[concat(subscription().id, '/resourceGroups/', parameters('resourceGroupName'), '/providers/Microsoft.Network/frontdoorwebapplicationfirewallpolicies/', parameters('wafPolicyName'))]"
+            ],
+            "properties": {
+                "friendlyName": "app-eoc-dashboard-frontend",
```
[the remark this answers is not in the document]
Comment (echappen) [reply iac/arm-templates/front-door-app-service.json]: whoops, fixed

File: iac/create-metrics-resources.bash (hunk 1 above)
Comment (echappen) [reply iac/create-metrics-resources.bash]: Resolved [here](https://github.com/18F/piipan/pull/507/commits/8a95c4a775f2f152e14762e5e26fd2cd561ac979#diff-3f860f483e93933e27df096b2bbeea9cd9a0252ad92c104cb073c9cf5740b5dfR74)

File: iac/create-resources.bash (hunk 2 above)
Comment (echappen) [reply iac/create-resources.bash]: Resolved [here](https://github.com/18F/piipan/pull/507/commits/8a95c4a775f2f152e14762e5e26fd2cd561ac979#diff-3f860f483e93933e27df096b2bbeea9cd9a0252ad92c104cb073c9cf5740b5dfR74)

File: iac/create-metrics-resources.bash (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanhofdotgov) [reply iac/create-metrics-resources.bash]: Ah, of course.

File: iac/add-front-door-to-app.bash (hunk 4 above)
[the remark this answers is not in the document]
Comment (ryanhofdotgov) [reply iac/add-front-door-to-app.bash]: @kingcomma As I don't have access at the moment, could you poke around in the government portal and see what Azure Front Door might use for a domain there?

**Review by ryanhofdotgov — APPROVED** [review]

**Pull request merged**